### ISSUE TITLE
Unsolved A11y issues

### ISSUE THREAD

**Issue by zwiastunsw** [joomla/accessibility#54]
### 1. Alerts are not accessible for colour blind
See: [joomla-alert [a11y] #99](https://github.com/joomla-projects/custom-elements/issues/99)
@BrianTeeman closed it for the following reason:

> the alerts need a massive rewrite as they are being used completely wrong. It's not if this code is accessible or not but it is where it is being used in the cms. A modal alert is a style it is not necessarily a role=alert - they are not the same thing at all

### 2. Toolbar - accessibility
See: [Toolbar - About accessibility #42](https://github.com/joomla/accessibility/issues/42)

The toolbar is fully accessible to mouse users (and sighted users), but not to keyboard and blind users. There is no way to access the toolbar quickly and directly.
After executing the selected action, the  keyboard and blind users does not return to the place from which the action was called. The reason is the unimplemented keyboard support.
See topic below.

### ~3. Efficient navigation on the Backend page for keyboard users~

See: [[a11y] Provide efficient navigation on the Backend page for keyboard users #442](https://github.com/joomla/40-backend-template/issues/442)
@BrianTeeman proposed a solution, but closed PR. See: [[4.0][a11y] SkipTo Links for Atum #22149](https://github.com/joomla/joomla-cms/pull/22149)
@BrianTeeman merged  [[4.0][a11y] SkipTo Links for Atum #24142] (https://github.com/joomla/joomla-cms/pull/24142)

### 4. Admin table - Table option

See: [Report: Table option #45](https://github.com/joomla/accessibility/issues/45)

The problem is partially solved by @brianteeman.
The PR was successfully tested, but we did not carry out any verification tests. See: [[4.0] [a11y] Table Options #22008](https://github.com/joomla/joomla-cms/pull/22008)

### 5. Admin table:  Drag & drop function is not accessible

[Explore of Backend pages type "manager" at Joomla 4 #40](https://github.com/joomla/accessibility/issues/40)
To change the order of items, the administrator can use drag & drop. Unfortunately, users who only use the keyboard cannot perform this task.

### <del>6. Admin table: improve accessibility</del>

[Explore of Backend pages type "manager" at Joomla 4 #40](https://github.com/joomla/accessibility/issues/40)

@BrianTeeman presented the concept of a solution. Unfortunately, he closed PR due to lack of tests. 
See: [[4.0] [a11y] proof of concept for admin tables #21977](https://github.com/joomla/joomla-cms/pull/21977)
Merged. See: [[4.0] Sortable Table [a11y] #22404](https://github.com/joomla/joomla-cms/pull/22404) (thx @Brianteeman)

### 7. No information on the level of nesting of subcategories
[Explore of Backend pages type "manager" at Joomla 4 #40](https://github.com/joomla/accessibility/issues/40)
In the category table, the information about the level of nesting of the subcategory is important. Unfortunately, this information is not passed on to screen readers at all.

### 8. Links to labels
See: [[4.0][a11y] Links to buttons #23737](https://github.com/joomla/joomla-cms/issues/23737#issuecomment-459867017)
This needs to be checked. It will probably be repaired.

**Comment by brianteeman** [comment]
6. This was merged as #22404

**Comment by brianteeman** [comment]
7. No information on the level of nesting of subcategories

Do you have any examples of how to address this. I looked and I can't find anything. And I assume it applies to menu items as well

**Comment by brianteeman** [comment]
Are 2 and 3 resolved by the skipto plugin

**Comment by zwiastunsw** [comment]
@brianteeman : I do not know of the example of a 'pure' solution. I know that you can create tree views in a table using JQuery scripts , e.g.:  [Free jQuery tree table Plugins](https://www.jqueryscript.net/tags.php?/tree%20table/). I will be looking for it.

**Comment by brianteeman** [comment]
1. A collapsible tree view such as those is not appropriate
2. I dont see any accessible advantage with that approach

**Comment by brianteeman** [comment]
Presumably this is also an issue 
![image](https://user-images.githubusercontent.com/1296369/54686980-69eac980-4b12-11e9-9116-88ebede61420.png)
